subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/EclipseModelBuilder.java

```diff
@@ -365,14 +365,22 @@ public boolean isSatisfiedBy(DefaultEclipseProject element) {
     private static DefaultAccessRule createAccessRule(AccessRule accessRule) {
         int kindCode;
         String kind = accessRule.getKind();
-        if (kind.equals("accessible") || kind.equals("0")) {
-            kindCode = 0;
-        } else if (kind.equals("nonaccessible") || kind.equals("1")) {
-            kindCode = 1;
-        } else if (kind.equals("discouraged") || kind.equals("2")) {
-            kindCode = 2;
-        } else {
-            kindCode = 0;
+        switch (kind) {
+            case "accessible":
```

Comment:
I am not a big fan of the fall-through here. I wonder if the if/else code above was better. WDYT?

Reply:
Fall-through, when we're executing code before going to the next label is really confusing, but this is basically just a list, right?
E.g. @lptr explicitly requested such fall-throughs in my other pr: https://github.com/gradle/gradle-groovy-all/commit/0c71e540f2084d00e99a411be5147ff3cf9293d4#diff-c197962302397baf3a4cc36463dce5eaL85

Reply:
So our guide-line is that fall-through is OK when the code for the two cases is identical? Fine with me.